code/game/objects/items/stacks/stack.dm

```diff
@@ -55,11 +55,10 @@
 	. = ..()
 	if(!dynamic_icon_state)
 		return
-	var/temp_amount = get_amount()
-	if(temp_amount > 1)
-		icon_state = "[initial(icon_state)]_[min(temp_amount, 3)]" //2 if amount is 2, 3 if more.
-		return
-	icon_state = initial(icon_state)
+	var/state = CEILING((amount/max_amount) * 3, 1)
+	icon_state = "[initial(icon_state)]_[state]"
+	if(state <= 1)
+		icon_state = initial(icon_state)
```

Comment:
```suggestion
	icon_state = state <= 1 ? initial(icon_state) : "[initial(icon_state)]_[state]"
```
I would go rather with that

Reply:
@Contrabang Readable enough, what do you think?

Reply:
That would work, but I don't think its readable. Ternaries are more difficult to read then if statements imo